api/routes/upload_refactored.py

```
@@ -0,0 +1,112 @@
+        )
+    except Exception as e:
+        logger.error(f"Failed to upload chunk: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
+
+@user_router.post("/finalize")
+async def finalize_upload(request: UploadFinalizeRequest) -> Dict[str, Any]:
+    """Finalize upload and assemble chunks"""
+    try:
+        # Handle both camelCase and snake_case
+        upload_id = request.upload_id or request.uploadId
+        total_chunks = request.total_chunks or request.totalChunks
```

Ruff (0.12.2)

Local variable `total_chunks` is assigned to but never used

Remove assignment to unused variable `total_chunks`

(F841)